src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -760,6 +771,21 @@ def __init__(self, config):
         # Initialize weights and apply final processing
         self.post_init()
 
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.PathLike]], *model_args, **kwargs):
```

Comment:
Uff that's quite hacky here - not a big fan of overwriting methods such as `from_pretrained(...)` :-/ I think I'd be more in favor of adding a:

```python
if model.config.force_lm_head_in_fp32 and model.lm_head.weight.dtype != torch.float32:
    model.lm_head.to(torch.float32)
```

to the forward call of the model. IMO better than adding a new `from_pretrained(...)` and a `post_init()` - wdyt?


Also just to understand

Reply:
I think the forward is okay as well but it feels like the wrong callback, this means that everything we add after the `cast_fo_fp32` when setting up the model is going to have to be in forward ...

Well the flow I want is `tie weights -> cast lm_head to fp32` that should happen more or less in the same callback (typically I thought that `post_init` was enough, but turns out `from_pretrained` ignores the weight tying in `post_init` to do it itself, if you could explain why I'd be interested @patrickvonplaten )

Reply:
Actually I think I've found a better way: [fec68bf](https://github.com/huggingface/transformers/pull/18141/commits/fec68bf1187e635468f9e1aac83030ee7bf40e27)